conda-store-server/conda_store_server/build.py

```diff
@@ -110,6 +111,99 @@ def build_environment(conda_command, environment_filename, conda_prefix):
     )
 
 
+def build_lock_environment(lock_filename, conda_prefix):
+    return subprocess.check_output(
+        ["conda-lock", "install", "--prefix", conda_prefix, lock_filename],
+        stderr=subprocess.STDOUT,
+        encoding="utf-8",
+    )
+
+
+def fetch_and_extract_packages(conda_store, lock_filename):
+    """Fetch links from a conda-locked build recipe and then
+    gets a filelock on the required folder.
+    """
+    import tarfile
+    import shutil
+    from conda import CONDA_SOURCE_ROOT
+
+    extensions = {".tar": "r:", ".tar.gz": "r:gz", ".tar.bz2": "r:bz2"}
+
+    root = pathlib.Path(CONDA_SOURCE_ROOT).parts[0:3]
+    prefix = pathlib.Path(*root, "pkgs")
+
+    try:
+        prefix.exists()
+    except FileNotFoundError as e:
+        conda_store.log.error(
+            f"The conda prefix {prefix} does not exist. Traceback: {e}"
+        )
+
+    spec = dict()
+
+    with open(pathlib.Path(lock_filename)) as f:
+        spec = yaml.safe_load(f)
+
+    packages_searched = 1
+
+    for p in spec["package"]:
+        if p["manager"] != "conda":
+            # ignore non-conda managed packages for now
+            pass
+
+        else:
+            url = p["url"]
+
+            filename = pathlib.Path(url.split("/")[-1:][0])
+            filepath = prefix.joinpath(filename)
+            lockfile = filelock.FileLock(f"{filepath.__str__()}.lock")
+
+            num_files_log = f"{packages_searched} of {len(spec['package'])}"
```

Comment:
Could this just be directly added to the log statement below so we don't have to hunt down the line it updates?

Reply:
done!